tests/unit/test_solvers/test_casadi_algebraic_solver.py

```diff
@@ -97,10 +116,96 @@ def test_solve_with_input(self):
 
         # Solve
         solver = pybamm.CasadiAlgebraicSolver()
-        solution = solver.solve(model, np.linspace(0, 1, 10), inputs={"value": 7})
+        solution = solver.solve(model, np.linspace(0, 1, 10), inputs={"param": 7})
         np.testing.assert_array_equal(solution.y, -7)
 
 
+class TestCasadiAlgebraicSolverSensitivity(unittest.TestCase):
+    def test_solve_with_symbolic_input(self):
+        # Simple system: a single algebraic equation
+        var = pybamm.Variable("var")
+        model = pybamm.BaseModel()
+        model.algebraic = {var: var + pybamm.InputParameter("param")}
+        model.initial_conditions = {var: 2}
+        model.variables = {"var": var}
+
+        # create discretisation
+        disc = pybamm.Discretisation()
+        disc.process_model(model)
+
+        # Solve
+        solver = pybamm.CasadiAlgebraicSolver()
+        solution = solver.solve(model, [0], inputs={"param": "[sym]"})
+        np.testing.assert_array_equal(solution["var"].value(7), -7)
```

Comment:
Do you test the case where the input to value is a dict? I couldn't see this on a quick skim through the code

Reply:
Checked this in the test for `ProcessedCasadiVariable` but can add a test here too